R/slide.R

```diff
@@ -133,33 +122,30 @@ epi_slide = function(x, f, ..., n, ref_time_values,
     ref_time_values = ref_time_values[ref_time_values %in%
                                       unique(x$time_value)] 
   }
-              
-  # If before is missing, then use align to set up alignment
+  
+  # Before cannot be missing if after is set to 0. If after is set to a nonzero
+  # number, then before must be set to 0
   if (missing(before)) {
-    align = match.arg(align)
-    if (align == "right") {
-      before_num = n-1
-      after_num = 0
-    }
-    else if (align == "center") {
-      before_num = floor((n-1)/2)
-      after_num = ceiling((n-1)/2)
-    }
-    else {
-      before_num = 0
-      after_num = n-1
+    if (after == 0) {
+      Abort("`before` cannot be missing when `after` is set to 0.")
+    } else {
+      Warn("`before` missing but `after` nonzero;
+           `before` has been set to 0.")
+      before = 0 
     }
   }
   
   # Otherwise set up alignment based on passed before value
-  else {
-    if (before < 0 || before > n-1) {
-      Abort("`before` must be in between 0 and n-1`.")
-    }
-
-    before_num = before
-    after_num = n-1-before
+  if (before < 0 || after < 0) {
+    Abort("`before` and `after` must be at least 0.")
   }
+  
+  if (floor(before) < ceiling(before) || floor(after) < ceiling(after)) {
+    Abort("`before` and `after` must be integers.")
+  }
+  
+  before_num = before
+  after_num = after
```

Comment:
Can we collapse the code in this area to something like the following?
```r
  if (missing(time_step)) {
    before_num = before
    after_num = after
  } else {
    before_num = time_step(before_num)
    after_num = time_step(after_num)
  }
```

Reply:
I just replaced `before_num` with `before` and `after_num` with `after` as I deemed this redundant.